commit 55197c9b6f53865dcff4e52ab9e8974e49177465 (refs/changes/61/1661/4)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-03-14T18:26:24+00:00 (2 years, 7 months ago)
    
    DOCKER-1017 sdc-docker cli-commit.js test failing with 'invalid reference format'

diff --git a/test/integration/cli-commit.test.js b/test/integration/cli-commit.test.js
index 4d36b5e..16286e9 100644
--- a/test/integration/cli-commit.test.js
+++ b/test/integration/cli-commit.test.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 /*
@@ -33,6 +33,7 @@ var STATE = {
 
 var CONTAINER_PREFIX = 'sdcdockertest_commit_';
 var IMAGE_NAME = 'busybox';
+var IMAGE_PREFIX = 'sdcdockertest-commit';
 var TP = 'api: commit: ';  // Test prefix.
 
 test(TP + 'setup', function (tt) {
@@ -50,7 +51,7 @@ test(TP + 'setup', function (tt) {
 
 test(TP + 'test add file', function (tt) {
 
-    var commitImageTag = common.makeContainerName(CONTAINER_PREFIX);
+    var commitImageTag = common.makeImageName(IMAGE_PREFIX);
     var containerName = common.makeContainerName(CONTAINER_PREFIX);
 
     tt.test('run ' + IMAGE_NAME + ' container', function (t) {
diff --git a/test/lib/common.js b/test/lib/common.js
index 7acb0b6..b139a06 100644
--- a/test/lib/common.js
+++ b/test/lib/common.js
@@ -285,6 +285,14 @@ function makeContainerName(prefix) {
 }
 
 
+/*
+ * Make a prefixed, randomized name for a test image.
+ */
+function makeImageName(prefix) {
+    return prefix + '-' + libuuid.create().split('-')[0];
+}
+
+
 /*
  * Parse docker columnar output by using the widths in the first header row.
  */
@@ -347,6 +355,7 @@ module.exports = {
     expCliErr: expCliErr,
     ifErr: ifErr,
     makeContainerName: makeContainerName,
+    makeImageName: makeImageName,
     objCopy: objCopy,
     parseOutputUsingHeader: parseOutputUsingHeader,
     partialExp: partialExp
